Source/Urho3D/IO/File.cpp

```diff
@@ -440,75 +567,18 @@ bool File::OpenInternal(const ea::string& fileName, FileMode mode, bool fromPack
         return false;
     }
 
-#ifdef __ANDROID__
-    if (URHO3D_IS_ASSET(fileName))
-    {
-        if (mode != FILE_READ)
-        {
-            URHO3D_LOGERROR("Only read mode is supported for Android asset files");
-            return false;
-        }
-
-        assetHandle_ = SDL_RWFromFile(URHO3D_ASSET(fileName), "rb");
-        if (!assetHandle_)
-        {
-            URHO3D_LOGERRORF("Could not open Android asset file %s", fileName.c_str());
-            return false;
-        }
-        else
-        {
-            name_ = fileName;
-            absoluteFileName_ = fileName;
-            mode_ = mode;
-            position_ = 0;
-            if (!fromPackage)
-            {
-                size_ = SDL_RWsize(assetHandle_);
-                offset_ = 0;
-            }
-            checksum_ = 0;
-            return true;
-        }
-    }
-#endif
+    // For an encoded chunked stream we don't know actual data size in the underlying file so let's not limit it.
+    unsigned size = (packageEntry && !encoded) ? packageEntry->size_ : ea::numeric_limits<unsigned>::max();
```

Comment:
Why is this a limitation? Is there any reason we can not always save original file size in package entry?